Project-Service-implementation-Script/main.py

```diff
@@ -0,0 +1,5238 @@
+    # global fileheader
+    fileheader = ["Resource","Process","Status","Remark"]
+
+    if not path.exists(file_exists):
+        with open(file_exists, 'w', newline='',encoding='utf-8') as file:
+            writer = csv.writer(file, quoting=csv.QUOTE_NONNUMERIC, delimiter=',',lineterminator='\n')
+            writer.writerows([fileheader])
+    with open(file_exists, 'a', newline='',encoding='utf-8') as file:
+        writer = csv.writer(file, quoting=csv.QUOTE_NONNUMERIC, delimiter=',',lineterminator='\n')
+        writer.writerows([messageArr])
+
+def checkEmailValidation(email):
+    if (re.search(regex, email)):
+        return True
+    else:
+        return False
+
+# Fetch user details 
+def fetchUserDetails(environment, accessToken, dikshaId):
+    global OrgName
+    url = config.get(environment, 'elevateuserhost') + config.get(environment, 'userInfoApiUrl')
+    messageArr = ["User search API called."]
+    headers = {'Content-Type': 'application/json',
+               'internal-access-token': config.get(environment, 'internal-access-token'),
+               'x-authenticated-user-token': accessToken}
+    isEmail = checkEmailValidation(dikshaId.lstrip().rstrip())
+    if isEmail:
+        body = "{\n  \"request\": {\n    \"filters\": {\n    \t\"email\": \"" + dikshaId.lstrip().rstrip() + "\"\n    },\n      \"fields\" :[],\n    \"limit\": 1000,\n    \"sort_by\": {\"createdDate\": \"desc\"}\n  }\n}"
+    else:
+        body = "{\n  \"request\": {\n    \"filters\": {\n    \t\"userName\": \"" + dikshaId.lstrip().rstrip() + "\"\n    },\n      \"fields\" :[],\n    \"limit\": 1000,\n    \"sort_by\": {\"createdDate\": \"desc\"}\n  }\n}"
+
+    responseUserSearch = requests.request("POST", url, headers=headers, data=body)
+    if responseUserSearch.status_code == 200:
+        responseUserSearch = responseUserSearch.json()
+        if responseUserSearch['result']['response']['content']:
+            userKeycloak = responseUserSearch['result']['response']['content'][0]['userId']
+            userName = responseUserSearch['result']['response']['content'][0]['userName']
+            firstName = responseUserSearch['result']['response']['content'][0]['firstName']
+            rootOrgId = responseUserSearch['result']['response']['content'][0]['rootOrgId']
+            for index in responseUserSearch['result']['response']['content'][0]['organisations']:
+                if rootOrgId == index['organisationId']:
+                    roledetails = index['roles']
+                    rootOrgName = index['orgName']
+                    OrgName.append(index['orgName'])
+            print(roledetails)
+        else:
+            terminatingMessage("-->Given username/email is not present in DIKSHA platform<--.")
+    else:
+        terminatingMessage("User fetch API failed. Check logs.")
+    return [userKeycloak, userName, firstName,roledetails,rootOrgName,rootOrgId]
+
+
+# fetch org Ids 
+def fetchOrgId(environment, accessToken, parentFolder, OrgName):
```

Comment:
@MallanagoudaB is this function used any where. If yes can you please point out the line number where it is used